internal/auth/ldap/repository_auth_method_read.go

```diff
@@ -0,0 +1,235 @@
+	if len(scopeIds) == 0 {
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing scope IDs")
+	}
+	authMethods, err := r.getAuthMethods(ctx, "", scopeIds, opt...)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op)
+	}
+	return authMethods, nil
+}
+
+// lookupAuthMethod will lookup a single auth method
+func (r *Repository) lookupAuthMethod(ctx context.Context, authMethodId string, opt ...Option) (*AuthMethod, error) {
+	const op = "ldap.(Repository).lookupAuthMethod"
+	var err error
+	ams, err := r.getAuthMethods(ctx, authMethodId, nil, opt...)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op)
+	}
+	switch {
+	case len(ams) == 0:
+		return nil, nil // not an error to return no rows for a "lookup"
+	case len(ams) > 1:
+		return nil, errors.New(ctx, errors.NotSpecificIntegrity, op, fmt.Sprintf("%s matched more than 1 ", authMethodId))
+	default:
+		return ams[0], nil
+	}
+}
+
+// getAuthMethods allows the caller to either lookup a specific AuthMethod via
+// its id or search for a set AuthMethods within a set of scopes.  Passing both
+// scopeIds and a authMethod is an error. The WithUnauthenticatedUser,
+// WithLimit and WithOrder options are supported and all other options are
+// ignored.
+//
+// The AuthMethod returned has its value objects populated (SigningAlgs,
+// CallbackUrls, AudClaims and Certificates).  The AuthMethod returned has its
+// IsPrimaryAuthMethod bool set.
+//
+// When no record is found it returns nil, nil
+func (r *Repository) getAuthMethods(ctx context.Context, authMethodId string, scopeIds []string, opt ...Option) ([]*AuthMethod, error) {
+	const op = "ldap.(Repository).getAuthMethods"
+	if authMethodId == "" && len(scopeIds) == 0 {
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing search criteria: both auth method id and scope ids are empty")
+	}
+	if authMethodId != "" && len(scopeIds) > 0 {
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "searching for both an auth method id and scope ids is not supported")
+	}
+
+	const aggregateDelimiter = "|"
+
+	dbArgs := []db.Option{}
+	opts, err := getOpts(opt...)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op)
+	}
+	limit := r.defaultLimit
+	if opts.withLimit != 0 {
+		// non-zero signals an override of the default limit for the repo.
+		limit = opts.withLimit
+	}
+	dbArgs = append(dbArgs, db.WithLimit(limit))
+
+	if opts.withOrderByCreateTime {
+		if opts.ascending {
+			dbArgs = append(dbArgs, db.WithOrder("create_time asc"))
+		} else {
+			dbArgs = append(dbArgs, db.WithOrder("create_time"))
+		}
+	}
+
+	var args []any
+	var where []string
+	switch {
+	case authMethodId != "":
+		where, args = append(where, "public_id = ?"), append(args, authMethodId)
+	default:
+		where, args = append(where, "scope_id in(?)"), append(args, scopeIds)
+	}
+
+	if opts.withUnauthenticatedUser {
+		where, args = append(where, "state = ?"), append(args, string(ActivePublicState))
```

Comment:
Why does this imply this `where` statement? Could you add a small comment here?

Reply:
I'll add a comment.